test_setup.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# this one is to get everything set up without a hitch so you can run your
+# ansible command
+
+USERNAME=$USER
+
+# get ansible-conda down if it is not already
+git submodule init
+git submodule update
+
+# ssl stuff for nginx
+openssl genrsa -out security/ssl.key 1024 > /dev/null 2>&1
+openssl req -new -key security/ssl.key -out security/ssl.csr > /dev/null 2>&1 \
+<< EOF
+US
+CA
+Davis
+Libretexts
+
+${USERNAME}
+
+
+
+EOF
+openssl x509 -req -in security/ssl.csr -signkey security/ssl.key \
+	-out security/ssl.crt
+
+# cookie secret
+openssl rand -hex 32 > security/cookie_secret
+
+# generate ssh keys
+ssh-keygen >> /dev/null 2>&1 << EOF
```

Comment:
Comments need more explanation for what this all does.

Reply:
addressed in 6206f4015d63ef6bd96bcc302f07a6145713476c